plugins/queue/smtp_forward.js

```diff
@@ -339,9 +341,10 @@ exports.get_mx = function (next, hmail, domain) {
         return next(OK, plugin.get_mx_next_hop(hmail.todo.notes.next_hop));
     }
 
-    if (domain !== domain.toLowerCase()) domain = domain.toLowerCase();
+    const dom = plugin.cfg.main.domain_selector === 'mail_from' ? hmail.todo.mail_from.host.toLowerCase() : domain.toLowerCase();
```

Comment:
```suggestion
    let dom = domain.toLowerCase();
    if (plugin.cfg.main.domain_selector && plugin.cfg.main.domain_selector === 'mail_from') hmail.todo.mail_from.host.toLowerCase();
```

Should we check that `domain_selector` is defined first?

Reply:
FWIW that suggestion is broken.

Reply:
Not that it matters, but I've fixed it.